packages/webpack/src/plugin.js

```diff
@@ -221,11 +225,38 @@ class LavaMoatPlugin {
           JAVASCRIPT_MODULE_TYPE_ESM,
         ]
 
+        // Webpack has a concept of ignored modules
+        // When a module is ignored a carveout is necessary in policy enforcement for it because the ID that webpack creates for it is not exactly helpful.
+        // example outcome in the bundle: `const nodeCrypto = __webpack_require__(/*! crypto */ "?0b7d");`
+        // Sadly, even treeshaking doesn't eliminate that module. It's left there and failing to work when reached by runtime policy enforcement.
+        // Below is the most reliable way I've found to date to identify ignored modules.
+        /**
+         * @param {import('webpack').Module} m
+         * @returns {boolean}
+         */
+        const isIgnoredModule = (m) => {
+          if (
+            m.type === JAVASCRIPT_MODULE_TYPE_DYNAMIC &&
+            // @ts-expect-error BAD TYPES
+            m.identifierStr?.startsWith('ignored')
+          ) {
+            return true
+          }
+          // This function must be very narrow, because it's gatekeeping policy generation too
+          return false
+        }
+
         /**
          * @param {import('webpack').Module} m
-         * @returns {m is import('webpack').NormalModule}
+         * @param {string} moduleClass
+         * @returns {m is import('webpack').NormalModule} // TODO: this is not
+         *   true anymore, but there's no superclass of all reasonable module
+         *   types
          */
-        const isNormalModule = (m) => 'resource' in m
+        const isInspectableModule = (m, moduleClass) =>
+          'userRequest' in m ||
+          m.type?.startsWith('javascript') ||
+          ['ExternalModule'].includes(moduleClass)
```

Comment:
Alright, why is this not `moduleClass === 'ExternalModule'`?  this had better be good 😄